wazuh/wazuh
# Change FIM scan to use DBSync transactions.

|Related issue|
|---|
|#10847|


## Description
Hello team!! 

This PR aims to make the necessary changes in the FIM scan code in order to use DBSync transactions.

Closes #10847 


## Manual testing
I  have performed the following tests to check if the behavior is correct or not:
- Monitor a folder and set the scan frequency to 10 seconds.
- Create a number of files.
- Wait until the scan raises the added events.
- Modify the files
- Wait until the scan raises the modified events.
- Delete all the files
- Wait until the scan raises the deleted events.


# Tests

- Compilation without warnings in every supported platform


- Memory tests for Linux
- Memory tests for Windows
